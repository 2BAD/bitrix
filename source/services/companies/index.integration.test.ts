/* tslint-env jest */
// tslint:disable: no-expression-statement object-literal-sort-keys

import Bitrix from './../../bitrix'

const WEBHOOK_URL = process.env.WEBHOOK_URL

// tslint:disable-next-line: no-if-statement
if (!WEBHOOK_URL) {
  // tslint:disable-next-line: no-throw
  throw Error('Integration tests require environmental variable `WEBHOOK_URL` to be set')
}

const { companies } = Bitrix(WEBHOOK_URL)

describe('Companies', () => {
  describe('fields', () => {
    it('should get all fields', async () => {
      const { result } = await companies.fields()
      expect(result).toMatchSnapshot()
    })
  })
})
