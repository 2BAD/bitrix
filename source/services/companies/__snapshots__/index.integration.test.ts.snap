// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies fields should get all fields 1`] = `
Object {
  "ADDRESS": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Фактический адрес",
    "type": "string",
  },
  "ADDRESS_2": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Адрес (стр. 2)",
    "type": "string",
  },
  "ADDRESS_CITY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Город",
    "type": "string",
  },
  "ADDRESS_COUNTRY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Страна",
    "type": "string",
  },
  "ADDRESS_COUNTRY_CODE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Код страны",
    "type": "string",
  },
  "ADDRESS_LEGAL": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Юридический адрес",
    "type": "string",
  },
  "ADDRESS_POSTAL_CODE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Почтовый индекс",
    "type": "string",
  },
  "ADDRESS_PROVINCE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Область",
    "type": "string",
  },
  "ADDRESS_REGION": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Район",
    "type": "string",
  },
  "ASSIGNED_BY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Ответственный",
    "type": "user",
  },
  "BANKING_DETAILS": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Банковские реквизиты",
    "type": "string",
  },
  "COMMENTS": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Комментарий",
    "type": "string",
  },
  "COMPANY_TYPE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "statusType": "COMPANY_TYPE",
    "title": "Тип компании",
    "type": "crm_status",
  },
  "CONTACT_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": true,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Контакт",
    "type": "crm_contact",
  },
  "CREATED_BY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Кем создана",
    "type": "user",
  },
  "CURRENCY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Валюта",
    "type": "crm_currency",
  },
  "DATE_CREATE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Дата создания",
    "type": "datetime",
  },
  "DATE_MODIFY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Дата изменения",
    "type": "datetime",
  },
  "EMAIL": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": true,
    "isReadOnly": false,
    "isRequired": false,
    "title": "E-mail",
    "type": "crm_multifield",
  },
  "EMPLOYEES": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "statusType": "EMPLOYEES",
    "title": "Кол-во сотрудников",
    "type": "crm_status",
  },
  "HAS_EMAIL": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Задан e-mail",
    "type": "char",
  },
  "HAS_IMOL": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Задана открытая линия",
    "type": "char",
  },
  "HAS_PHONE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Задан телефон",
    "type": "char",
  },
  "ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "ID",
    "type": "integer",
  },
  "IM": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": true,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Мессенджер",
    "type": "crm_multifield",
  },
  "INDUSTRY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "statusType": "INDUSTRY",
    "title": "Сфера деятельности",
    "type": "crm_status",
  },
  "IS_MY_COMPANY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Моя компания",
    "type": "char",
  },
  "LEAD_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Лид",
    "type": "crm_lead",
  },
  "LOGO": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Логотип",
    "type": "file",
  },
  "MODIFY_BY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Кем изменена",
    "type": "user",
  },
  "OPENED": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Доступна для всех",
    "type": "char",
  },
  "ORIGINATOR_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Внешний источник",
    "type": "string",
  },
  "ORIGIN_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Идентификатор элемента во внешнем источнике",
    "type": "string",
  },
  "ORIGIN_VERSION": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Версия оригинала",
    "type": "string",
  },
  "PHONE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": true,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Телефон",
    "type": "crm_multifield",
  },
  "REG_ADDRESS": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Юридический адрес",
    "type": "string",
  },
  "REG_ADDRESS_2": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Юридический адрес (стр. 2)",
    "type": "string",
  },
  "REG_ADDRESS_CITY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Юридический адрес город",
    "type": "string",
  },
  "REG_ADDRESS_COUNTRY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Юридический адрес страна",
    "type": "string",
  },
  "REG_ADDRESS_COUNTRY_CODE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Юридический адрес код страны",
    "type": "string",
  },
  "REG_ADDRESS_POSTAL_CODE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Юридический адрес почтовый индекс ",
    "type": "string",
  },
  "REG_ADDRESS_PROVINCE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Юридический адрес область",
    "type": "string",
  },
  "REG_ADDRESS_REGION": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Юридический адрес район",
    "type": "string",
  },
  "REVENUE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Годовой оборот",
    "type": "double",
  },
  "TITLE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": true,
    "title": "Название компании",
    "type": "string",
  },
  "UTM_CAMPAIGN": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Обозначение рекламной кампании",
    "type": "string",
  },
  "UTM_CONTENT": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Содержание кампании",
    "type": "string",
  },
  "UTM_MEDIUM": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Тип трафика",
    "type": "string",
  },
  "UTM_SOURCE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Рекламная система",
    "type": "string",
  },
  "UTM_TERM": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Условие поиска кампании",
    "type": "string",
  },
  "WEB": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": true,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Сайт",
    "type": "crm_multifield",
  },
}
`;
